jcache/embedded/src/test/java/org/infinispan/jcache/JCacheConfigurationTest.java

```diff
@@ -40,18 +53,96 @@ public void testJCacheManagerWherePathContainsFileSchemaAndAbsolutePath() throws
-      URI uriWithJarFileSchema = URI.create(uri.toString().replace("file", "jar:file"));
+   public class MyClassLoader extends URLClassLoader {
+      MyClassLoader(URL[] urls, ClassLoader parent) {
+         super(urls, parent);
+      }
 
-      withCachingProvider(provider -> {
-         JCacheManager jCacheManager = new JCacheManager(
-               uriWithJarFileSchema,
-               provider.getClass().getClassLoader(),
-               provider,
-               null);
-         assertTrue(null != jCacheManager.getCache("foo"));
-      });
+      public void addURL(URL url) {
+         super.addURL(url);
+      }
+   }
+
+   public void testJCacheManagerWithRealJarFileSchema() throws Exception {
+      File sampleJarWithResourceFile = null;
+      ClassLoader originalClassLoader = null;
+
+      try {
+         // given - creating sample jar file
+         sampleJarWithResourceFile = new File("sampleJarWithResource.jar");
+         String existingResourceToAddInJar = "infinispan_uri.xml";
+         String targetPathInJar = "sample-dir-inside-jar/";
+         createJar(sampleJarWithResourceFile, existingResourceToAddInJar, targetPathInJar);
+
+         // given - adding jar file into classpath
+         String fullTargetPath = targetPathInJar + existingResourceToAddInJar;
+
+         originalClassLoader = Thread.currentThread().getContextClassLoader();
+         MyClassLoader myClassLoader = new MyClassLoader(((URLClassLoader)originalClassLoader).getURLs(), originalClassLoader);
+         myClassLoader.addURL(new URI("jar:" + sampleJarWithResourceFile.toURI().toString() + "!/").toURL());
+         Thread.currentThread().setContextClassLoader(myClassLoader);
+
+         // when
+         URI resourceInsideJarUri = new URI("jar:" + sampleJarWithResourceFile.toURI().toString() + "!/" + fullTargetPath);
+         withCachingProvider(provider -> {
+            JCacheManager jCacheManager = new JCacheManager(
+                resourceInsideJarUri,
+                provider.getClass().getClassLoader(),
+                provider,
+                null);
+
+            // then
+            assertTrue(null != jCacheManager.getCache("foo"));
+         });
+      }
+      finally {
+         // cleanup
+         if (sampleJarWithResourceFile != null && sampleJarWithResourceFile.exists()) {
+            sampleJarWithResourceFile.delete();
+         }
+
+         if (originalClassLoader != null) {
+            Thread.currentThread().setContextClassLoader(originalClassLoader);
+         }
+      }
+   }
+
+   private void createJar(File jarFile, String existingResourceToAddInJar, String targetPathInJar) throws IOException, URISyntaxException {
+      Manifest manifest = new Manifest();
+      manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+      JarOutputStream sampleJarOutputStream = new JarOutputStream(new FileOutputStream(jarFile), manifest);
```

Comment:
For the test, the jar file should be created in the tmp directory. See org.infinispan.test.TestingUtil#tmpDirectory(java.lang.Class<?>).

Reply:
Ok, now test tmp folder is used.